src/mongocxx/instance.hpp

```diff
@@ -49,10 +50,17 @@ class MONGOCXX_API instance {
     ///
     ~instance();
 
+    ///
+    /// Sets a user-provided @c logger for the driver to use.
+    ///
+    /// @param
+    ///  logger The logger that the driver will direct log messages to.
+    ///
+    void set_logger(std::unique_ptr<logger> logger);
```

Comment:
What is the thread-safety level for this function? What happens if someone calls set_logger(a), then calls set_logger(b) from a's operator()?